core/arch/arm/kernel/generic_boot.c

```diff
@@ -760,21 +760,25 @@ static uint64_t get_dt_val_and_advance(const void *data, size_t *offs,
 static int add_res_mem_dt_node(struct dt_descriptor *dt, const char *name,
 			       paddr_t pa, size_t size)
 {
-	int offs;
-	int ret;
-	int addr_size = 2;
-	int len_size = 2;
+	int offs = 0;
+	int ret = 0;
+	int addr_size = -1;
+	int len_size = -1;
 	char subnode_name[80];
 
 	offs = fdt_path_offset(dt->blob, "/reserved-memory");
-	if (offs >= 0) {
-		addr_size = fdt_address_cells(dt->blob, offs);
-		if (addr_size < 0)
-			return -1;
-		len_size = fdt_size_cells(dt->blob, offs);
-		if (len_size < 0)
-			return -1;
-	} else {
+
+	if (offs < 0)
+		offs = 0;
+
+        len_size = fdt_size_cells(dt->blob, offs);
```

Comment:
So at this point there is no difference between `fdt_path_offset()` returning 0 and returning `< 0`. Yet in the first case the node exists, in the second case it doesn't. In the first case we'll get into troubles when trying to add it again, and we will return an error.

Is it safe to assume that we won't find `/reserved-memory` at offset 0?

Reply:
Not looking into the libfdt, it might be possible that the reserved memory node is at the beginning of the device tree and this equals offset 0. I reintroduced the found from your initial suggestion to handle this case.